types/shares_test.go

```diff
@@ -456,6 +456,41 @@ func TestContigShareWriter(t *testing.T) {
 	assert.Equal(t, txs, resTxs)
 }
 
+func TestSplitMessage(t *testing.T) {
```

Comment:
while this test is a nice to have, most of the testing is done by the round trip tests https://github.com/celestiaorg/celestia-core/blob/843c996acd2dc544db68f9fa870d4e7be3f3af63/types/shares_test.go#L218